Index: src/main/java/ru/mtsbank/hw3/Main.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/Main.java b/src/main/java/ru/mtsbank/hw3/Main.java
deleted file mode 100644
--- a/src/main/java/ru/mtsbank/hw3/Main.java	
+++ /dev/null	
@@ -1,11 +0,0 @@
-package ru.mtsbank.hw3;
-
-import ru.mtsbank.hw3.AnimalService.CreateAnimalServiceImpl;
-
-public class Main {
-    public static void main(String[] args) {
-       CreateAnimalServiceImpl service = new CreateAnimalServiceImpl();
-       service.createAnimal();
-       service.createAnimal(5);
-    }
-}
Index: src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalServiceImpl.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalServiceImpl.java b/src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalServiceImpl.java
deleted file mode 100644
--- a/src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalServiceImpl.java	
+++ /dev/null	
@@ -1,73 +0,0 @@
-package ru.mtsbank.hw3.AnimalService;
-
-import ru.mtsbank.hw3.animal.fish.models.Barbus;
-import ru.mtsbank.hw3.animal.fish.models.GoldFish;
-import ru.mtsbank.hw3.animal.fish.models.Guppy;
-import ru.mtsbank.hw3.animal.herbivores.models.Cow;
-import ru.mtsbank.hw3.animal.pet.models.Cat;
-import ru.mtsbank.hw3.animal.pet.models.Dog;
-import ru.mtsbank.hw3.animal.pet.models.Hamster;
-import ru.mtsbank.hw3.animal.predator.models.Lion;
-import ru.mtsbank.hw3.animal.predator.models.Shark;
-import ru.mtsbank.hw3.animal.predator.models.Wolf;
-
-import java.math.BigDecimal;
-
-public class CreateAnimalServiceImpl implements CreateAnimalService {
-
-    public void createAnimal(int number) {
-        for(int i = 0; i < number; i++){
-            int remains = i%10;
-            switch (remains) {
-                case 0:
-                    System.out.println(new Cat("Чиширская", "Барсик",
-                            new BigDecimal(123.5698), "Агресивный"));
-                    break;
-                case 1:
-                    System.out.println(new Dog("Дворняга","Шарик",
-                            new BigDecimal(300.123),"Дружелюбный"));
-                    break;
-                case 2:
-                    System.out.println(new Hamster("Обыкновенный хомяк", "Кругляш",
-                            new BigDecimal(543), "Ленивый"));
-                    break;
-                case 3:
-                    System.out.println(new Barbus("Барбус Алый","Немо",
-                            new BigDecimal(1000.9999), "Нет"));
-                    break;
-                case 4:
-                    System.out.println( new Guppy("Гуппи", "Губка-Боп",
-                            new BigDecimal(10943.3232),"Ну такой"));
-                    break;
-                case 5:
-                    System.out.println(new GoldFish("Золотая-рыбка", "Чудо",
-                            new BigDecimal(3213.22),"Богатый" ));
-                    break;
-                case 6:
-                    System.out.println(new Shark("Белая", "Killer","Agresive"));
-                    break;
-                case 7:
-                    System.out.println(new Wolf("Simple", "Volk", "Killer"));
-                    break;
-                case 8:
-                    System.out.println(new Lion("Simle", "Korol", "Pain"));
-                    break;
-                case 9:
-                    System.out.println(new Cow("Red", "Burenka",
-                            new BigDecimal(200000.9872),"Pain"));
-                    break;
-            }
-
-        }
-    }
-
-    @Override
-    public void createAnimal() {
-        int i = 0;
-        do{
-            CreateAnimalService.super.createAnimal();
-            i++;
-        }
-        while (i < 10);
-        }
-    }
Index: src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalService.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalService.java b/src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalService.java
deleted file mode 100644
--- a/src/main/java/ru/mtsbank/hw3/AnimalService/CreateAnimalService.java	
+++ /dev/null	
@@ -1,75 +0,0 @@
-package ru.mtsbank.hw3.AnimalService;
-
-import ru.mtsbank.hw3.animal.fish.models.Barbus;
-import ru.mtsbank.hw3.animal.fish.models.GoldFish;
-import ru.mtsbank.hw3.animal.fish.models.Guppy;
-import ru.mtsbank.hw3.animal.herbivores.models.Cow;
-import ru.mtsbank.hw3.animal.pet.models.Cat;
-import ru.mtsbank.hw3.animal.pet.models.Dog;
-import ru.mtsbank.hw3.animal.pet.models.Hamster;
-import ru.mtsbank.hw3.animal.predator.models.Lion;
-import ru.mtsbank.hw3.animal.predator.models.Shark;
-import ru.mtsbank.hw3.animal.predator.models.Wolf;
-
-import java.math.BigDecimal;
-
-public interface CreateAnimalService {
-
-    /**
-     * Метод который проверяет
-     * к кому типу относиться животное
-     * */
-    default void checkTypeAnimal(int num){
-        int i = 0;
-        while (i < num) {
-            int remains = i % 10;
-            switch (remains) {
-                case 0:
-                    System.out.println(new Cat("Чиширская", "Барсик",
-                            new BigDecimal(123.5698), "Агресивный"));
-                    break;
-                case 1:
-                    System.out.println(new Dog("Дворняга","Шарик",
-                            new BigDecimal(300.123),"Дружелюбный"));
-                    break;
-                case 2:
-                    System.out.println(new Hamster("Обыкновенный хомяк", "Кругляш",
-                            new BigDecimal(543), "Ленивый"));
-                    break;
-                case 3:
-                    System.out.println(new Barbus("Барбус Алый","Немо",
-                            new BigDecimal(1000.9999), "Нет"));
-                    break;
-                case 4:
-                    System.out.println( new Guppy("Гуппи", "Губка-Боп",
-                            new BigDecimal(10943.3232),"Ну такой"));
-                    break;
-                case 5:
-                    System.out.println(new GoldFish("Золотая-рыбка", "Чудо",
-                            new BigDecimal(3213.22),"Богатый" ));
-                    break;
-                case 6:
-                    System.out.println(new Shark("Белая", "Killer","Agresive"));
-                    break;
-                case 7:
-                    System.out.println(new Wolf("Simple", "Volk","Killer"));
-                    break;
-                case 8:
-                    System.out.println(new Lion("Simle", "Korol","Pain"));
-                    break;
-                case 9:
-                    System.out.println(new Cow("Red", "Burenka",
-                            new BigDecimal(200000.9872),"Pain"));
-                    break;
-            }
-            i++;
-        }
-    }
-
-    /**
-     * Метод по созданию 10 животных
-     * */
-    default void createAnimal() {
-        checkTypeAnimal(10);
-    }
-}
Index: src/main/java/ru/mtsbank/hw3/animal/AbstractAnimal.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/AbstractAnimal.java b/src/main/java/ru/mtsbank/hw3/animal/AbstractAnimal.java
deleted file mode 100644
--- a/src/main/java/ru/mtsbank/hw3/animal/AbstractAnimal.java	
+++ /dev/null	
@@ -1,53 +0,0 @@
-package ru.mtsbank.hw3.animal;
-
-import java.math.BigDecimal;
-import java.math.RoundingMode;
-
-public abstract class AbstractAnimal implements Animal {
-    protected String breed;
-    protected String name;
-    protected BigDecimal cost;
-    protected String character;
-
-    public AbstractAnimal(String breed, String name, BigDecimal cost, String character) {
-        this.breed = breed;
-        this.name = name;
-        this.cost= cost.setScale(2,RoundingMode.DOWN);
-        this.character = character;
-    }
-
-    public AbstractAnimal(String breed, String name, String character) {
-        this.breed = breed;
-        this.name = name;
-        this.character = character;
-    }
-
-    public String getBreed() {
-        return breed;
-    }
-
-    @Override
-    public String getName() {
-        return name;
-    }
-
-    @Override
-    public BigDecimal getCost() {
-        return cost;
-    }
-
-    @Override
-    public String getCharacter() {
-        return character;
-    }
-
-    @Override
-    public String toString() {
-        return "AbstractAnimal{" +
-                "breed='" + breed + '\'' +
-                ", name='" + name + '\'' +
-                ", cost=" + cost +
-                ", character='" + character + '\'' +
-                '}';
-    }
-}
Index: src/main/java/ru/mtsbank/hw/animalservice/AnimalTypes.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/animalservice/AnimalTypes.java b/src/main/java/ru/mtsbank/hw/animalservice/AnimalTypes.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/animalservice/AnimalTypes.java	
@@ -0,0 +1,14 @@
+package ru.mtsbank.hw.animalservice;
+
+public enum AnimalTypes {
+    BARBUS,
+    GOLDFISH,
+    GUPPY,
+    COW,
+    CAT,
+    DOG,
+    HAMSTER,
+    LION,
+    SHARK,
+    WOLF,
+}
Index: src/main/java/ru/mtsbank/hw/animalservice/CreateAnimal.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimal.java b/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimal.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimal.java	
@@ -0,0 +1,63 @@
+package ru.mtsbank.hw.animalservice;
+
+import ru.mtsbank.hw.animal.AbstractAnimal;
+import ru.mtsbank.hw.animal.fish.models.Barbus;
+import ru.mtsbank.hw.animal.fish.models.GoldFish;
+import ru.mtsbank.hw.animal.fish.models.Guppy;
+import ru.mtsbank.hw.animal.herbivores.models.Cow;
+import ru.mtsbank.hw.animal.pet.models.Cat;
+import ru.mtsbank.hw.animal.pet.models.Dog;
+import ru.mtsbank.hw.animal.pet.models.Hamster;
+import ru.mtsbank.hw.animal.predator.models.Lion;
+import ru.mtsbank.hw.animal.predator.models.Shark;
+import ru.mtsbank.hw.animal.predator.models.Wolf;
+
+import java.math.BigDecimal;
+
+public class CreateAnimal {
+    public AbstractAnimal create(AnimalTypes types){
+        AbstractAnimal animal;
+        switch (types) {
+            case CAT:
+                animal = new Cat("Чиширская", "Барсик",
+                        new BigDecimal(123.5698), "Агресивный");
+                break;
+            case DOG:
+                animal = new Dog("Дворняга","Шарик",
+                        new BigDecimal(300.123),"Дружелюбный");
+                break;
+            case HAMSTER:
+                animal = new Hamster("Обыкновенный хомяк", "Кругляш",
+                        new BigDecimal(543), "Ленивый");
+                break;
+            case BARBUS:
+                animal = new Barbus("Барбус Алый","Немо",
+                        new BigDecimal(1000.9999), "Нет");
+                break;
+            case GUPPY:
+                animal = new Guppy("Гуппи", "Губка-Боп",
+                        new BigDecimal(10943.3232),"Ну такой");
+                break;
+            case GOLDFISH:
+                animal = new GoldFish("Золотая-рыбка", "Чудо",
+                        new BigDecimal(3213.22),"Богатый" );
+                break;
+            case SHARK:
+                animal = new Shark("Белая", "Killer","Agresive");
+                break;
+            case WOLF:
+                animal = new Wolf("Simple", "Volk", "Killer");
+                break;
+            case LION:
+                animal = new Lion("Simle", "Korol", "Pain");
+                break;
+            case COW:
+                animal = new Cow("Red", "Burenka",
+                        new BigDecimal(200000.9872),"Pain");
+                break;
+            default:
+                throw new IllegalArgumentException("Wrong type: " + types);
+        }
+        return animal;
+    }
+}
Index: src/main/java/ru/mtsbank/hw/searchservise/SearchService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/searchservise/SearchService.java b/src/main/java/ru/mtsbank/hw/searchservise/SearchService.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/searchservise/SearchService.java	
@@ -0,0 +1,20 @@
+package ru.mtsbank.hw.searchservise;
+
+import ru.mtsbank.hw.animal.AbstractAnimal;
+
+public interface SearchService {
+    /**
+     * Метод находит всех животных которые родились в високосный год
+     * */
+
+    AbstractAnimal[] findLeapYearNames(AbstractAnimal[] animals);
+
+    /**
+     * Метод находит всех животных старших N лет
+     * */
+    AbstractAnimal[] findOlderAnimal(AbstractAnimal[] animals, int N);
+    /**
+     * Вывод дупликатов животных
+     * */
+    void findDuplicate(AbstractAnimal[] animals);
+}
Index: src/main/java/ru/mtsbank/hw/searchservise/SearchServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/searchservise/SearchServiceImpl.java b/src/main/java/ru/mtsbank/hw/searchservise/SearchServiceImpl.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/searchservise/SearchServiceImpl.java	
@@ -0,0 +1,45 @@
+package ru.mtsbank.hw.searchservise;
+
+import ru.mtsbank.hw.animal.AbstractAnimal;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SearchServiceImpl implements SearchService{
+
+    @Override
+    public AbstractAnimal[] findLeapYearNames(AbstractAnimal[] animals) {
+        List<AbstractAnimal> list = new ArrayList<>();
+        for(int i = 0; i < animals.length; i++){
+            if(animals[i].getBirthDate() != null) {
+                if (animals[i].getBirthDate().isLeapYear()) {
+                    list.add(animals[i]);
+                }
+            }
+        }
+        AbstractAnimal[] abstractAnimals = new AbstractAnimal[list.size()];
+        return list.toArray(abstractAnimals);
+    }
+
+    @Override
+    public AbstractAnimal[] findOlderAnimal(AbstractAnimal[] animals, int N) {
+        List<AbstractAnimal> list = new ArrayList<>();
+        for(int i = 0; i < animals.length; i++){
+            if(animals[i].getBirthDate() != null) {
+                if (LocalDate.now().getYear() - animals[i].getBirthDate().getYear() > N) {
+                    list.add(animals[i]);
+                }
+            }
+        }
+        AbstractAnimal[] abstractAnimals = new AbstractAnimal[list.size()];
+        return list.toArray(abstractAnimals);
+    }
+
+    @Override
+    public void findDuplicate(AbstractAnimal[] animals) {
+        for(int i = 0; i < animals.length; i++){
+            System.out.println(animals.clone());
+        }
+    }
+}
Index: src/main/java/ru/mtsbank/hw/animal/AbstractAnimal.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/animal/AbstractAnimal.java b/src/main/java/ru/mtsbank/hw/animal/AbstractAnimal.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/animal/AbstractAnimal.java	
@@ -0,0 +1,89 @@
+package ru.mtsbank.hw.animal;
+
+import javax.swing.text.DateFormatter;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+
+public abstract class AbstractAnimal implements Animal {
+    protected String breed;
+    protected String name;
+    protected BigDecimal cost;
+    protected String character;
+    protected LocalDate birthDate;
+
+    public AbstractAnimal(String breed, String name, BigDecimal cost, String character) {
+        this.breed = breed;
+        this.name = name;
+        this.cost= cost.setScale(2,RoundingMode.DOWN);
+        this.character = character;
+        this.birthDate = randomDate();
+    }
+
+    public AbstractAnimal(String breed, String name, String character) {
+        this.breed = breed;
+        this.name = name;
+        this.character = character;
+        this.birthDate = randomDate();
+    }
+
+    private LocalDate randomDate(){
+        long betweenDays = ChronoUnit.DAYS.between(LocalDate.of(1900,1,1),
+                LocalDate.now());
+        long random = ThreadLocalRandom.current().nextLong(betweenDays);
+        return LocalDate.of(1900,1,1).plusDays(random);
+    }
+
+    public String getBreed() {
+        return breed;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public BigDecimal getCost() {
+        return cost;
+    }
+
+    @Override
+    public String getCharacter() {
+        return character;
+    }
+
+    @Override
+    public LocalDate getBirthDate() {
+        return birthDate;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        AbstractAnimal that = (AbstractAnimal) o;
+        return Objects.equals(breed, that.breed) && Objects.equals(name, that.name) && Objects.equals(cost, that.cost) && Objects.equals(character, that.character) && Objects.equals(birthDate, that.birthDate);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(breed, name, cost, character, birthDate);
+    }
+
+    @Override
+    public String toString() {
+        return "AbstractAnimal{" +
+                "breed='" + breed + '\'' +
+                ", name='" + name + '\'' +
+                ", cost=" + cost +
+                ", character='" + character + '\'' +
+                ", birthDate=" + birthDate.format(DateTimeFormatter.ofPattern("dd-MM-yyyy")) +
+                '}';
+    }
+}
Index: src/main/java/ru/mtsbank/hw/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/Main.java b/src/main/java/ru/mtsbank/hw/Main.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/Main.java	
@@ -0,0 +1,17 @@
+package ru.mtsbank.hw;
+
+import ru.mtsbank.hw.animalservice.CreateAnimalServiceImpl;
+import ru.mtsbank.hw.searchservise.SearchService;
+import ru.mtsbank.hw.searchservise.SearchServiceImpl;
+
+import java.util.Arrays;
+
+public class Main {
+    public static void main(String[] args) {
+        CreateAnimalServiceImpl createAnimalService = new CreateAnimalServiceImpl();
+        SearchService service = new SearchServiceImpl();
+        System.out.println(Arrays.toString(service.findLeapYearNames(createAnimalService.createAnimal(23))));
+        System.out.println(Arrays.deepToString(service.findOlderAnimal(createAnimalService.createAnimal(),5)));
+        service.findDuplicate(createAnimalService.createAnimal());
+    }
+}
Index: src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalService.java b/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalService.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalService.java	
@@ -0,0 +1,28 @@
+package ru.mtsbank.hw.animalservice;
+
+import ru.mtsbank.hw.animal.AbstractAnimal;
+
+import java.util.Random;
+
+public interface CreateAnimalService {
+
+    /**
+     * Метод который проверяет
+     * к кому типу относиться животное
+     * */
+    default AbstractAnimal[] checkTypeAnimal(int num){
+        CreateAnimal createAnimal = new CreateAnimal();
+        AbstractAnimal[] abstractAnimals = new AbstractAnimal[num];
+        int i = 0;
+        while(i < num){
+            abstractAnimals[i] = createAnimal.create(AnimalTypes.values()
+                    [new Random().nextInt(AnimalTypes.values().length)]);
+            i++;
+        }
+        return abstractAnimals;
+        }
+
+        default AbstractAnimal[] createAnimal(){
+        return checkTypeAnimal(10);
+        }
+    }
Index: src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalServiceImpl.java b/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalServiceImpl.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/ru/mtsbank/hw/animalservice/CreateAnimalServiceImpl.java	
@@ -0,0 +1,21 @@
+package ru.mtsbank.hw.animalservice;
+
+import ru.mtsbank.hw.animal.AbstractAnimal;
+
+import java.util.Random;
+
+public class CreateAnimalServiceImpl implements CreateAnimalService {
+    @Override
+    public AbstractAnimal[] createAnimal() {
+        return CreateAnimalService.super.createAnimal();
+    }
+    public AbstractAnimal[] createAnimal(int num) {
+        CreateAnimal createAnimal = new CreateAnimal();
+        AbstractAnimal[] abstractAnimals = new AbstractAnimal[num];
+        for(int i = 0; i < num; i++){
+            abstractAnimals[i] = createAnimal.create(AnimalTypes.values()
+                    [new Random().nextInt(AnimalTypes.values().length)]);
+        }
+        return abstractAnimals;
+    }
+}
Index: gradle/wrapper/gradle-wrapper.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
new file mode 100644
--- /dev/null	
+++ b/gradle/wrapper/gradle-wrapper.properties	
@@ -0,0 +1,7 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
+networkTimeout=10000
+validateDistributionUrl=true
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
