Index: src/main/java/ru/mtsbank/hw3/animal/pet/models/Cat.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/pet/models/Cat.java b/src/main/java/ru/mtsbank/hw/animal/pet/models/Cat.java
rename from src/main/java/ru/mtsbank/hw3/animal/pet/models/Cat.java
rename to src/main/java/ru/mtsbank/hw/animal/pet/models/Cat.java
--- a/src/main/java/ru/mtsbank/hw3/animal/pet/models/Cat.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/pet/models/Cat.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.pet.models;
+package ru.mtsbank.hw.animal.pet.models;
 
-import ru.mtsbank.hw3.animal.pet.Pet;
+import ru.mtsbank.hw.animal.pet.Pet;
 
 import java.math.BigDecimal;
 
Index: src/main/java/ru/mtsbank/hw3/animal/pet/models/Dog.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/pet/models/Dog.java b/src/main/java/ru/mtsbank/hw/animal/pet/models/Dog.java
rename from src/main/java/ru/mtsbank/hw3/animal/pet/models/Dog.java
rename to src/main/java/ru/mtsbank/hw/animal/pet/models/Dog.java
--- a/src/main/java/ru/mtsbank/hw3/animal/pet/models/Dog.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/pet/models/Dog.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.pet.models;
+package ru.mtsbank.hw.animal.pet.models;
 
-import ru.mtsbank.hw3.animal.pet.Pet;
+import ru.mtsbank.hw.animal.pet.Pet;
 
 import java.math.BigDecimal;
 
@@ -8,4 +8,5 @@
     public Dog(String bread, String name, BigDecimal cost, String character) {
         super(bread, name, cost, character);
     }
+
 }
Index: src/main/java/ru/mtsbank/hw3/animal/herbivores/Herbivores.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/herbivores/Herbivores.java b/src/main/java/ru/mtsbank/hw/animal/herbivores/Herbivores.java
rename from src/main/java/ru/mtsbank/hw3/animal/herbivores/Herbivores.java
rename to src/main/java/ru/mtsbank/hw/animal/herbivores/Herbivores.java
--- a/src/main/java/ru/mtsbank/hw3/animal/herbivores/Herbivores.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/herbivores/Herbivores.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.herbivores;
+package ru.mtsbank.hw.animal.herbivores;
 
-import ru.mtsbank.hw3.animal.AbstractAnimal;
+import ru.mtsbank.hw.animal.AbstractAnimal;
 
 import java.math.BigDecimal;
 
Index: src/main/java/ru/mtsbank/hw3/animal/fish/models/GoldFish.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/fish/models/GoldFish.java b/src/main/java/ru/mtsbank/hw/animal/fish/models/GoldFish.java
rename from src/main/java/ru/mtsbank/hw3/animal/fish/models/GoldFish.java
rename to src/main/java/ru/mtsbank/hw/animal/fish/models/GoldFish.java
--- a/src/main/java/ru/mtsbank/hw3/animal/fish/models/GoldFish.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/fish/models/GoldFish.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.fish.models;
+package ru.mtsbank.hw.animal.fish.models;
 
-import ru.mtsbank.hw3.animal.fish.Fish;
+import ru.mtsbank.hw.animal.fish.Fish;
 
 import java.math.BigDecimal;
 
Index: src/main/java/ru/mtsbank/hw3/animal/fish/models/Guppy.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/fish/models/Guppy.java b/src/main/java/ru/mtsbank/hw/animal/fish/models/Guppy.java
rename from src/main/java/ru/mtsbank/hw3/animal/fish/models/Guppy.java
rename to src/main/java/ru/mtsbank/hw/animal/fish/models/Guppy.java
--- a/src/main/java/ru/mtsbank/hw3/animal/fish/models/Guppy.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/fish/models/Guppy.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.fish.models;
+package ru.mtsbank.hw.animal.fish.models;
 
-import ru.mtsbank.hw3.animal.fish.Fish;
+import ru.mtsbank.hw.animal.fish.Fish;
 
 import java.math.BigDecimal;
 
@@ -8,4 +8,5 @@
     public Guppy(String breed, String name, BigDecimal cost, String character) {
         super(breed, name, cost, character);
     }
+
 }
Index: src/main/java/ru/mtsbank/hw3/animal/pet/models/Hamster.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/pet/models/Hamster.java b/src/main/java/ru/mtsbank/hw/animal/pet/models/Hamster.java
rename from src/main/java/ru/mtsbank/hw3/animal/pet/models/Hamster.java
rename to src/main/java/ru/mtsbank/hw/animal/pet/models/Hamster.java
--- a/src/main/java/ru/mtsbank/hw3/animal/pet/models/Hamster.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/pet/models/Hamster.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.pet.models;
+package ru.mtsbank.hw.animal.pet.models;
 
-import ru.mtsbank.hw3.animal.pet.Pet;
+import ru.mtsbank.hw.animal.pet.Pet;
 
 import java.math.BigDecimal;
 
@@ -8,4 +8,5 @@
     public Hamster(String breed, String name, BigDecimal cost, String character) {
         super(breed, name, cost, character);
     }
+
 }
Index: src/main/java/ru/mtsbank/hw3/animal/fish/models/Barbus.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/fish/models/Barbus.java b/src/main/java/ru/mtsbank/hw/animal/fish/models/Barbus.java
rename from src/main/java/ru/mtsbank/hw3/animal/fish/models/Barbus.java
rename to src/main/java/ru/mtsbank/hw/animal/fish/models/Barbus.java
--- a/src/main/java/ru/mtsbank/hw3/animal/fish/models/Barbus.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/fish/models/Barbus.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.fish.models;
+package ru.mtsbank.hw.animal.fish.models;
 
-import ru.mtsbank.hw3.animal.fish.Fish;
+import ru.mtsbank.hw.animal.fish.Fish;
 
 import java.math.BigDecimal;
 
Index: src/main/java/ru/mtsbank/hw3/animal/predator/models/Wolf.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/predator/models/Wolf.java b/src/main/java/ru/mtsbank/hw/animal/predator/models/Wolf.java
rename from src/main/java/ru/mtsbank/hw3/animal/predator/models/Wolf.java
rename to src/main/java/ru/mtsbank/hw/animal/predator/models/Wolf.java
--- a/src/main/java/ru/mtsbank/hw3/animal/predator/models/Wolf.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/predator/models/Wolf.java	
@@ -1,8 +1,6 @@
-package ru.mtsbank.hw3.animal.predator.models;
+package ru.mtsbank.hw.animal.predator.models;
 
-import ru.mtsbank.hw3.animal.predator.Predator;
-
-import java.math.BigDecimal;
+import ru.mtsbank.hw.animal.predator.Predator;
 
 public class Wolf extends Predator {
 
Index: src/main/java/ru/mtsbank/hw3/animal/herbivores/models/Cow.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/herbivores/models/Cow.java b/src/main/java/ru/mtsbank/hw/animal/herbivores/models/Cow.java
rename from src/main/java/ru/mtsbank/hw3/animal/herbivores/models/Cow.java
rename to src/main/java/ru/mtsbank/hw/animal/herbivores/models/Cow.java
--- a/src/main/java/ru/mtsbank/hw3/animal/herbivores/models/Cow.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/herbivores/models/Cow.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.herbivores.models;
+package ru.mtsbank.hw.animal.herbivores.models;
 
-import ru.mtsbank.hw3.animal.herbivores.Herbivores;
+import ru.mtsbank.hw.animal.herbivores.Herbivores;
 
 import java.math.BigDecimal;
 
@@ -9,4 +9,6 @@
     public Cow(String breed, String name, BigDecimal cost, String character) {
         super(breed, name, cost, character);
     }
+
+
 }
Index: src/main/java/ru/mtsbank/hw3/animal/predator/models/Lion.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/predator/models/Lion.java b/src/main/java/ru/mtsbank/hw/animal/predator/models/Lion.java
rename from src/main/java/ru/mtsbank/hw3/animal/predator/models/Lion.java
rename to src/main/java/ru/mtsbank/hw/animal/predator/models/Lion.java
--- a/src/main/java/ru/mtsbank/hw3/animal/predator/models/Lion.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/predator/models/Lion.java	
@@ -1,11 +1,10 @@
-package ru.mtsbank.hw3.animal.predator.models;
+package ru.mtsbank.hw.animal.predator.models;
 
-import ru.mtsbank.hw3.animal.predator.Predator;
-
-import java.math.BigDecimal;
+import ru.mtsbank.hw.animal.predator.Predator;
 
 public class Lion extends Predator {
     public Lion(String breed, String name, String character) {
         super(breed, name, character);
     }
+
 }
Index: src/main/java/ru/mtsbank/hw3/animal/predator/models/Shark.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/predator/models/Shark.java b/src/main/java/ru/mtsbank/hw/animal/predator/models/Shark.java
rename from src/main/java/ru/mtsbank/hw3/animal/predator/models/Shark.java
rename to src/main/java/ru/mtsbank/hw/animal/predator/models/Shark.java
--- a/src/main/java/ru/mtsbank/hw3/animal/predator/models/Shark.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/predator/models/Shark.java	
@@ -1,8 +1,6 @@
-package ru.mtsbank.hw3.animal.predator.models;
+package ru.mtsbank.hw.animal.predator.models;
 
-import ru.mtsbank.hw3.animal.predator.Predator;
-
-import java.math.BigDecimal;
+import ru.mtsbank.hw.animal.predator.Predator;
 
 public class Shark extends Predator {
     public Shark(String breed, String name, String character) {
Index: src/main/java/ru/mtsbank/hw3/animal/Animal.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/Animal.java b/src/main/java/ru/mtsbank/hw/animal/Animal.java
rename from src/main/java/ru/mtsbank/hw3/animal/Animal.java
rename to src/main/java/ru/mtsbank/hw/animal/Animal.java
--- a/src/main/java/ru/mtsbank/hw3/animal/Animal.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/Animal.java	
@@ -1,6 +1,7 @@
-package ru.mtsbank.hw3.animal;
+package ru.mtsbank.hw.animal;
 
 import java.math.BigDecimal;
+import java.time.LocalDate;
 
 public interface Animal {
     /**
@@ -19,4 +20,9 @@
      * Функция возвращения поля{@link AbstractAnimal#character}
      * */
     String getCharacter();
+
+    /**
+     * Функция возвращает поля{@link AbstractAnimal#birthDate}
+     * */
+    LocalDate getBirthDate();
 }
Index: src/main/java/ru/mtsbank/hw3/animal/pet/Pet.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/pet/Pet.java b/src/main/java/ru/mtsbank/hw/animal/pet/Pet.java
rename from src/main/java/ru/mtsbank/hw3/animal/pet/Pet.java
rename to src/main/java/ru/mtsbank/hw/animal/pet/Pet.java
--- a/src/main/java/ru/mtsbank/hw3/animal/pet/Pet.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/pet/Pet.java	
@@ -1,7 +1,7 @@
-package ru.mtsbank.hw3.animal.pet;
+package ru.mtsbank.hw.animal.pet;
 
 
-import ru.mtsbank.hw3.animal.AbstractAnimal;
+import ru.mtsbank.hw.animal.AbstractAnimal;
 
 import java.math.BigDecimal;
 
Index: src/main/java/ru/mtsbank/hw3/animal/predator/Predator.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/predator/Predator.java b/src/main/java/ru/mtsbank/hw/animal/predator/Predator.java
rename from src/main/java/ru/mtsbank/hw3/animal/predator/Predator.java
rename to src/main/java/ru/mtsbank/hw/animal/predator/Predator.java
--- a/src/main/java/ru/mtsbank/hw3/animal/predator/Predator.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/predator/Predator.java	
@@ -1,7 +1,5 @@
-package ru.mtsbank.hw3.animal.predator;
-import ru.mtsbank.hw3.animal.AbstractAnimal;
-
-import java.math.BigDecimal;
+package ru.mtsbank.hw.animal.predator;
+import ru.mtsbank.hw.animal.AbstractAnimal;
 
 public abstract class Predator extends AbstractAnimal {
     public Predator(String breed, String name, String character) {
Index: src/main/java/ru/mtsbank/hw3/animal/fish/Fish.java
===================================================================
diff --git a/src/main/java/ru/mtsbank/hw3/animal/fish/Fish.java b/src/main/java/ru/mtsbank/hw/animal/fish/Fish.java
rename from src/main/java/ru/mtsbank/hw3/animal/fish/Fish.java
rename to src/main/java/ru/mtsbank/hw/animal/fish/Fish.java
--- a/src/main/java/ru/mtsbank/hw3/animal/fish/Fish.java	
+++ b/src/main/java/ru/mtsbank/hw/animal/fish/Fish.java	
@@ -1,6 +1,6 @@
-package ru.mtsbank.hw3.animal.fish;
+package ru.mtsbank.hw.animal.fish;
 
-import ru.mtsbank.hw3.animal.AbstractAnimal;
+import ru.mtsbank.hw.animal.AbstractAnimal;
 
 import java.math.BigDecimal;
 
